### ISSUE TITLE
Change Request: Allow multiple parsers for the same files

### ISSUE THREAD

**Issue by tim-fabian** [eslint/eslint#17655]
### ESLint version

v8.51.0

### What problem do you want to solve?

ESlint currently only supports a single parser for the same files.
In my use case I need to use rules from both @angular-eslint/eslint-plugin-template and @html-eslint/eslint-plugin to lint all my html files.
They both require a custom parser.

```javascript
import angularTemplateParser from '@angular-eslint/template-parser';
// ...
languageOptions: {
    parser: angularTemplateParser
},
// ...
```

### What do you think is the correct solution?

I'd like ESLint to support defining an array of parsers that run on the same files:

```javascript
import angularTemplateParser from '@angular-eslint/template-parser';
import htmlParser from '@html-eslint/parser';
// ...
languageOptions: {
    parser: [angularTemplateParser, htmlParser]
},
// ...
```

Unknown/Invalid rules should only throw an error when they are unknown/invalid with either parser.

### Participation

- [ ] I am willing to submit a pull request for this change.

### Additional comments

If the parser config can be either a single parser or an array of parsers this would probably not be a breaking change and also keep the same performance for single parser configurations (as running multiple parsers on the same file will probably be a lot slower)

**Comment by nzakas** [comment]
Thanks for the suggestion. What you're asking for isn't possible or practical. ESLint is built upon the assumption that each file is parsed by exactly one parser, and changing that would be a monumental amount of effort to support an edge case.

I think a better solution would be to ask the `angular-eslint` folks to duplicate the rules you'd like from the `html-eslint` plugin.
